include/boost/url/impl/ipv6_address.ipp

```diff
@@ -59,9 +59,8 @@ struct h16
         std::uint16_t v;
         for(;;)
         {
-            auto d =
-                grammar::hexdig_value(*it);
-            if(d == -1)
+            char d{' '};
```

Comment:
why the space?

Reply:
This is related to the asan error we had before. It was complaining the variable was not being initialized. 

I've fixed some other things since then and made `hexdig_value` deterministic so this might not be an error anymore. 

I just pushed another commit to check if any compiler still complains about it.